src/DateInput/DateInput.jsx

```diff
@@ -0,0 +1,132 @@
+// Global Imports
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+
+// Local imports
+import { DEFAULT_CLASS as DEFAULT_LABEL_CLASS } from '../Label/Label';
+import { classBuilder } from '../utils/Utils';
+import './DateInput.scss';
+
+export const DEFAULT_CLASS = 'govuk-date-input';
+const DateInput = ({
+  id,
+  name,
+  classBlock,
+  classModifiers,
+  className,
+  error,
+  day,
+  month,
+  year,
+  ...attrs
+}) => {
+  const classes = classBuilder(classBlock, classModifiers, className);
+
+  const initialState = {
+    [name + '-day']: day || '',
+    [name + '-month']: month || '',
+    [name + '-year']: year || '',
+  }
+
+  const [date, setDate] = useState(initialState);
+
+  const onChange = (event) => {
+    setDate({...date, [event.target.name] : event.target.value})
```

Comment:
This isn't going to have quite the effect you're expecting... or, at least, it's not guaranteed to. To do this cleanly, you want the current state, which is achieved by passing a function to `setDate`, where the function receives the current state:
```suggestion
    setDate(prev => ({...prev, [event.target.name] : event.target.value}));
```

Basically, there's an asynchronous nature to this and if you fire `onChange` multiple times in rapid succession (such as someone typing quickly) with your code, you're not guaranteed to capture all the changes. This has caught me out countless times!

Reply:
After swapping to TextInputs I realised what I was doing here was unrequired so the useState is removed completely